database/dtb.ksy

```diff
@@ -0,0 +1,127 @@
+meta:
+  id: dtb
+  title: Flattened Devicetree Format
+  file-extension: dtb
+  application:
+    - Linux
+    - Das U-Boot
+  xref:
+    wikidata: Q16960371
+  license: CC0-1.0
+  encoding: ASCII
+  endian: be
+doc: |
+  Also referred to as Devicetree Blob (DTB). It is a flat
+  binary encoding of data (primarily devicetree data, although
+  other data is possible as well).
+
+  On Linux systems that support this the blobs can be accessed in
+  /sys/firmware/fdt :
+
+  - https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw
+
+  The encoding of strings used in the strings block and struct block is
+  actually a subset of ASCII:
+
+  https://github.com/devicetree-org/devicetree-specification/blob/master/source/chapter2-devicetree-basics.rst
+
+  Example files:
+
+  - https://github.com/qemu/qemu/tree/master/pc-bios
+doc-ref:
+  - https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3
+  - https://github.com/devicetree-org/devicetree-specification/blob/ba2aa67/source/flattened-format.rst
+  - https://elinux.org/images/f/f4/Elc2013_Fernandes.pdf
+seq:
+  - id: magic
+    -orig-id: magic
+    contents: [0xd0, 0x0d, 0xfe, 0xed]
+  - id: total_size
+    -orig-id: totalsize
+    type: u4
+  - id: ofs_structure_block
+    -orig-id: off_dt_struct
+    type: u4
+  - id: ofs_strings_block
+    -orig-id: off_dt_strings
+    type: u4
+  - id: ofs_memory_reservation_block
+    -orig-id: off_mem_rsvmap
+    type: u4
+  - id: version
+    -orig-id: version
+    type: u4
+  - id: last_compatible_version
+    -orig-id: last_comp_version
+    type: u4
+    valid:
+      max: version
+  - id: boot_cpuid_phys
+    -orig-id: boot_cpuid_phys
+    type: u4
+  - id: strings_block_size
+    -orig-id: size_dt_strings
+    type: u4
+  - id: structure_block_size
+    -orig-id: size_dt_struct
+    type: u4
+instances:
+  memory_reservation_block:
+    pos: ofs_memory_reservation_block
+    size: ofs_structure_block - ofs_memory_reservation_block
+  structure_block:
+    pos: ofs_structure_block
+    size: ofs_strings_block - ofs_structure_block
```

Comment:
Is there a reason why you don't use `len_structure_block` instead (counting on that the above suggestion `'s/structure_block_size/len_structure_block/g'` is already applied)?

Reply:
I think this is a brainfart.